### ISSUE TITLE
d_hellenic_pagan creation bug

### ISSUE THREAD

**Issue by jbelig** [xirsoi/ancientreligions#27]
Sorry for the new issue, I have no clue how this whole interface works and where to put these reports.
Anyway, I was trying something and stumbled upon a bug. I created the d_hellenic_pagan as byzantine emperor and suddenly I lost my successors...

**Comment by xirsoi** [comment]
Can you give me more details on this?

**Comment by jbelig** [comment]
I started a game with the byzantine emperor and console changed him to hellenic. I then changed religion of the holders of 4 holy sites and gave my emperor control of c_alexandria to raise religious authority. Now, there's no decision to created the title, it's done normally as you create every other duchy, so that's what I did.
My emperor has several kids and before creating this my firstborn was my successor. After the creation, I got the message that I have no heir and I don't see the successor icon under my portrait. Hovering over byzantine empire icon says unknown successor too...
I tried several different things cause it was working a week ago. Pulled the relevant decision from ck2plus and added the line active=no in the end of the history file of the duchy, same result. Tried to do it with another ruler, same. It reminded me why I decided not to choose computer coding at the university 20 years ago...

**Comment by jbelig** [comment]
I think I found it. It probably happened cause my son was not hellenic and though he changed with the event the next day this wasn't fixed. I tried to change him first and then activate the title and it worked...

**Comment by jbelig** [comment]
That was definitely the problem. I did it again while all my sons were different religion and got the same. My previous heir changed again with the event after a day, but he appeared as my successor (for both the empire and the pontificate) only after a reload.

**Comment by rquinio** [comment]
Was d_hellenic_pagan your primary title ?
Maybe creation should require king tier level to avoid issues.
If I remember well d_norse_pagan_reformed behavior, I think it's supposed to be a secondary title that follows the primary title as long as the heir has the correct religion.

**Comment by xirsoi** [comment]
I would venture to guess that, under normal circumstances, anyone powerful enough to create the pontifex maximus title should probably have already converted their court.

I think requiring King tier or higher will help though.

**Comment by jbelig** [comment]
No it wasn't the primary title, I was the byzantine emperor.
It's true though, under normal circumstances my court would be converted already. 
By the way, I think it looks better and much more immersive if you do that by a decision and not just normal duchy creation.

**Comment by jbelig** [comment]
I found a way to lock that.
In the creation conditions I added the line ROOT = { current_heir = { religion = hellenic_pagan } }
So it's impossible to create it if your heir isn't hellenic. It's a bit sloppy, but does the job.
